spacify-ui/src/components/rooms/Rooms.jsx

```diff
@@ -1,36 +1,129 @@
+    setPhoneNo("");
+  };
+
+  const handleSubscribe = () => {
+    setShowPhoneNumberInput(true);
+  };
+
+  const handlePhoneNoChange = (e) => {
+    setPhoneNo(e.target.value);
+  };
+
+  const handleConfirmSubscription = () => {
+    console.log(`Subscribing to WhatsApp for room: ${selectedRoom.roomDescription}`);
+    console.log(`Phone number: ${phoneNo}`);
+    setShowPhoneNumberInput(false);
+
+    console.log("Saved phone number:", savedPhoneNo);
+    console.log("Saved room ID:", savedRoomId);
+//    let userId = global.sessionStorage.getItem(USER_NAME_KEY);
+          let userId = "kshatris";              //hardcoded change later
+          const requestHeader = {
+              method: 'POST',
+              headers: { 'Content-Type': 'application/json' },
+              body: JSON.stringify({ userId: userId, roomId: selectedRoom.roomId, phoneNumber: phoneNo})
+          };
+          fetch(UPDATE_SUBSCRIBED_ROOMS_API, requestHeader)
+              .then((res) => res.json())
+              .then((data) => {
+                  setResponse({header: data.success ? "Subscribed!" : "Uh-Oh error", show: true, responseMessage: data.message});
+              });
+  };
+
   return (
     <div className="rooms-container">
       <h1 className="rooms-header">My Rooms</h1>
       <ul className="rooms-list">
-        {rooms.map(room => (
-          <li key={room.id} className="rooms-list-item">
-            <h3>{room.name}</h3>
-            <p>Type: {room.type}</p>
+        {rooms.map((room) => (
+          <li key={room.roomId} className="rooms-list-item">
+            <h3 onClick={() => handleRoomSelect(room)}>{room.roomDescription}</h3>
           </li>
         ))}
       </ul>
+
+      {selectedRoom && (
+        <div className="room-details-overlay">
+          <div className="room-details-form">
+            <div className="form-header">
+              <h2>Room Details</h2>
+              <button className="close-button" onClick={handleCloseForm}>
+                X
+              </button>
+            </div>
+            <p className="room-name">Name: {selectedRoom.roomDescription}</p>
+            <p className="room-type">Type: {selectedRoom.type}</p>
+
+            {!showPhoneNumberInput ? (
+              <button id="formButton" variant="primary" size="lg" onClick={handleSubscribe}>
+                Subscribe to Whatsapp
+              </button>
+            ) : (
+              <div>
+                <label htmlFor="phoneNo">Enter Phone Number:</label>
+                <input
+                  type="text"
+                  id="phoneNo"
+                  value={phoneNo}
+                  onChange={handlePhoneNoChange}
+                />
+                <button className="confirm-button" id="formButton" onClick={handleConfirmSubscription}>
+                  Confirm Subscription
+                </button>
+              </div>
+            )}
+
+            {!showPhoneNumberInput && (
+              <div className="form-buttons">
```

Comment:
the whatsapp popup and the subscribtion pop up should close when we are done subscribing. Also we need to add a toast saying that this has been done